activerecord/lib/active_record/connection_adapters/postgresql/referential_integrity.rb

```diff
@@ -8,36 +8,32 @@ def supports_disable_referential_integrity? # :nodoc:
 
         def disable_referential_integrity # :nodoc:
           if supports_disable_referential_integrity?
-            original_exception = nil
+            tables_constraints = execute(<<-SQL).values
+              SELECT table_name, constraint_name
+              FROM information_schema.table_constraints
+              WHERE constraint_type = 'FOREIGN KEY'
+              AND is_deferrable = 'NO'
+              AND table_name IN (#{tables.collect { |name| quote(name) }.join(",")})
+            SQL
 
-            begin
-              transaction(requires_new: true) do
-                execute(tables.collect { |name| "ALTER TABLE #{quote_table_name(name)} DISABLE TRIGGER ALL" }.join(";"))
-              end
-            rescue ActiveRecord::ActiveRecordError => e
-              original_exception = e
-            end
+            execute(
+              tables_constraints.collect { |table, constraint|
+                "ALTER TABLE #{quote_table_name(table)} ALTER CONSTRAINT #{constraint} DEFERRABLE"
```

Comment:
Thank you your feedback :star: 

I noticed that the syntax requires newest PG.
I'm investigating the way to alternate constraints on old PG version.

Probably we must check the PG version whether it can use `ALTER CONSTRAINT`.